entity-service/src/main/kotlin/com/egm/stellio/entity/service/EntityService.kt

```diff
@@ -79,33 +75,21 @@ class EntityService(
             Entity(id = expandedEntity.id, type = listOf(expandedEntity.type), contexts = expandedEntity.contexts)
         val entity = entityRepository.save(rawEntity)
 
-        // filter the unwanted entries and expand all attributes for easier later processing
-        val propertiesAndRelationshipsMap = expandedEntity.attributes.mapValues {
-            expandValueAsMap(it.value)
-        }
-
-        propertiesAndRelationshipsMap.filter { entry ->
-            isAttributeOfType(entry.value, NGSILD_RELATIONSHIP_TYPE)
-        }.forEach { entry ->
+        expandedEntity.relationships.forEach { entry ->
             val relationshipType = entry.key
             val objectId = getRelationshipObjectId(entry.value)
-            val objectEntity = entityRepository.findById(objectId)
 
-            if (!objectEntity.isPresent) {
+            if (!exists(objectId))

```

Comment:
The validation could be done before. What I think is if the last Relationship is targeting a non existant entity, we did all the processing for the other relationships for nothing, what do you think ?

Reply:
Good idea. I guess we could even reuse part of the logic used in batch creation?

Reply:
I thought about it but the code in batch creation is just checking against a list of entities given in parameter, I don't know if we can find a correlation and unify both.

Reply:
It seems like I could just call `EntitiesGraphBuilder.build(...)` https://github.com/stellio-hub/stellio-context-broker/blob/develop/entity-service/src/main/kotlin/com/egm/stellio/entity/util/EntitiesGraphBuilder.kt#L26 with my single entity?